TODO

```diff
@@ -142,16 +142,37 @@ Features:
 
 * expose MS_NOSYMFOLLOW in various places
 
-* ability to insert trusted configuration and secrets into the boot parameters
-  of a kernel booting in a VM or on baremetal some way, via TPM
-  protection. idea:
-  1. pass via /proc/bootconfig
-  2. for secrets: put secrets in node of /proc/bootconfig, decrypt them via
-     TPM early on in PID 1, put them in $CREDENTIAL_PATH logic
-  3. for config: put signed data in node /proc/booconfig, validate via TPM
-     early on in PID 1, put data into /run/bootconfig/ as individual files
-  4. boot loader/stub should pick these up automatically from the boot loader
-     file systems
+* allow passing creds into kernel when booting: in EFI stub, collect creds
+  files from ESP directory, generate CPIO archive on the fly from them, so that
+  they are dropped into /run/initramfs/creds/ and pass to kernel as additional
+  initrd. Then, use LoadCredentialEncrypted=foo:/run/initramfs/creds/foo to
+  load them.
+
+* make LoadCredential= automatically find credentials in /etc/creds,
+  /run/creds, … and so on, if path component is unqualified
+
+* teach LoadCredential=/LoadCredentialEncrypted= to load credentials from
+  kernel cmdline, maybe: LoadCredentialEncrypted=foobar:proc-cmdline:foobar
+
+* credentials system:
+  - acquire from kernel command line
+  - acquire from EFI variable?
```

Comment:
I like this idea a lot, I have to support a number of boards that don't have dedicated TPM's where it would be quite handy to be able to bind authentication credentials to motherboards, this could be used for a number of use cases such as:

- ensuring that one can't accidentally duplicate authentication credentials when cloning hard drives by forcing a regeneration of credentials if decryption fails due to a missing EFI variable
- simplify secure hard drive reuse/resale during upgrade cycles in cases where the hard disks are replaced with their associated motherboards being kept in operation
- providing a clean transition path to more secure TPM based credential storage

Reply:
would be delighted to review a patch for this

Reply:
Hmm, I wonder...would this actually be possible to do already by just redirecting `/var/lib/systemd/credentials.secret` to something in `/sys/firmware/efi/efivars`?

Not sure how best/where to implement this though, would we want to have this integrated with `systemd-boot` in some way or just pull it from efivars once booted?